rand_distr/src/dirichlet.rs

```diff
@@ -27,22 +27,22 @@ use alloc::{boxed::Box, vec, vec::Vec};
 /// use rand::prelude::*;
 /// use rand_distr::Dirichlet;
 ///
-/// let dirichlet = Dirichlet::new(&[1.0, 2.0, 3.0]).unwrap();
+/// let dirichlet = Dirichlet::new([1.0, 2.0, 3.0]).unwrap();
 /// let samples = dirichlet.sample(&mut rand::thread_rng());
 /// println!("{:?} is from a Dirichlet([1.0, 2.0, 3.0]) distribution", samples);
 /// ```
 #[cfg_attr(doc_cfg, doc(cfg(feature = "alloc")))]
 #[derive(Clone, Debug, PartialEq)]
 #[cfg_attr(feature = "serde1", derive(serde::Serialize, serde::Deserialize))]
-pub struct Dirichlet<F>
+pub struct Dirichlet<F, const N: usize>
```

Comment:
The output length is now baked into the type which is a breaking change. Seems acceptable.

It does make the existence of `new_with_size` questionable (given we can now write `new([alpha; size])`).

Reply:
I removed `new_with_size`.